app/models/item.rb

```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
-end
\ No newline at end of file
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def customers_purchased
```

Comment:
Think of your method naming. A method named `customers_purchased` to me, makes me think this method will return a list of customer objects who purchased this item. Consider naming this something that indicates we're getting a count or number back.

Reply:
Also, this method isn't tested in the model. Remember that all model methods should be tested in the model spec, including the AR associations.